### ISSUE TITLE
lock the use of the bed

### ISSUE THREAD

**Issue by BrunoMine** [hkzorman/advanced_npc#35]
It is possible to sleep in a bed while an NPC is already sleeping.

**Comment by NewbProgrammer101** [comment]
Yes, it is possible to sleep while NPC is sleeping. The NPC is not registered as a player in Minetest though.

@hkzorman , where are you?

**Comment by NewbProgrammer101** [comment]
There are still people who care about this mod.

**Comment by hkzorman** [comment]
@BrunoMine Yes, this is an issue.
I wouldn't find it so common for a player to go ahead and sleep on a bed with a NPC on it... but yes, it can happen. I guess I can try to overwrite the `beds` `on_rightclick` definition to avoid player usage if it is used. Maybe a variable `used` can be added at the MTG `beds` mod and this would solve the problem forever. I will see.

@NewbProgrammer101 I wouldn't like the NPCs to be registered as players, that I think might complicate stuff. A simple flag in the node metadata should do it.
Btw, thanks for asking and caring about the mod. Real life has kept me very busy, and I also have been partially distracted in building a new mod, but I still care for this mod and have big plans for it! So yes, this isn't abandoned, it will definetely keep going on!

**Comment by BrunoMine** [comment]
I also have great expectations for this mod, I would like you to improve documentation of the code and improve the semantics in general. A lot of this I'm trying to document, but the structuring still seems very confusing. This will allow me to hone your work and ease your tasks with this.
I'm having good results with him, soon I'll be posting.
